File: day01/not_quite_lisp.py
from sys import argv

script, filename = argv

with open(filename) as f :
    for line in f :
        fl = basement = 0
        for i, c in enumerate(line) :
            if c == "(" :
                fl += 1 
            elif c == ")" :
                fl -= 1
                if fl < 0 and basement == 0 :
                    basement = i + 1
            elif c == "\n" :
                print "floor = %d, basement = %d" % (fl, basement)
                fl = basement = 0

File: day01/not_quite_lisp.c
#include <stdio.h>

int main (int argc, char *argv[]) {
        int c;
        int floor = 0;
        int basement = 0;
        int i = 0;

        while ((c = getchar()) != EOF) {
                i++;
                switch (c) {
                case '(' :
                        floor++;
                        break;
                case ')' :
                        floor--;
                        if (floor < 0 && basement == 0) {
                                basement = i;
                        }
                        break;
                case '\n' :
                        printf("floor = %d, basement = %d\n", floor, basement);
                        floor = 0;
                        basement = 0;
                        i = 0;
                        break;
                }
        }
        return 0;
}
